### PR TITLE
Updated README

### LINKED ISSUE #151: document local dev environment
create documentation for the local dev environment, namely:

- pre-requisites (docker + docker-compose) 
- how to use (setup + teardown)
- how to maintain and change

info should be enough for any developer to simply clone the repo and get the environment running with a single command (assuming pre-requisites are fulfilled).

for maintenance do document:
- git server image used
- explain the need for infrastructure files to mirror production environment
- document possible pitfalls (i.e. production repo or branches changing names)
- how to access the git server's web UI
- how to change default ports if needed
- how to access the containerized DB via a sql client (i.e. dbeaver, datagrip)

### PR DESCRIPTION
**Pull request by grimord**
Closes #151

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,63 +9,93 @@
...
+- **Database**: (SQL_DATABASE in .env file)
+- **Authentication**: Database Native
+- **Username**: (SQL_USER in .env file)
+- **Password**: (SQL_PASSWORD in .env file)
 
-    $ ./gradlew build
-    
-Or using an IDE with gradle support such as [IntelliJ IDEA](https://www.jetbrains.com/idea/) or [Eclipse](https://www.eclipse.org/ide/)
+### Stopping and removing the environment
+If you are running in **attached** mode (without the `-d` flag) you can use `Ctrl + C` to stop all containers **gracefully** or, if you're in a hurry (or a container is not stopping on its own) you can press `Ctrl + C` to **force shutdown**.
```
Comment (rjcanto) [inline README.md]: If you are running in **attached** mode (without the `-d` flag) you can use `Ctrl + C` to stop all containers **gracefully** or, if you're in a hurry (or a container is not stopping on its own) you can press `Ctrl + C`  a second time to **force shutdown**.

Comment (grimord) [reply README.md]: Fixed

File: README.md [hunk 2]
```diff
@@ -9,63 +9,93 @@
...
+```
+docker-compose up
+```
+This mode will run all the application containers and bind **their output to your standard output**. If you want to run them in the **background** simply add the **detach** flag to the command:
 ```
-spring.datasource.url=jdbc:postgresql://127.0.0.1:5432/spring_batch
-spring.datasource.username=postgres
-spring.datasource.password=1234 
+docker-compose up -d
 ```
```
Comment (jtrindade) [inline README.md:54]: After bringing up the containers, it is possible to interact with i-on integration. Make it explicit here what are the ports used to access the git server or any other available service.

File: README.md [hunk 3]
```diff
@@ -9,63 +9,93 @@
...
 
-There is also the need to update `ion.core-base-url` and `ion.core-token`. Check [I-On Core documentation](https://github.com/i-on-project/core/blob/master/README.md)
+The **first time you build the images may take a while** but you'll notice following runs are much faster due to the way Docker caches its image layers.
```
Comment (jtrindade) [inline README.md]: The first time you build the images **it** may take a while

**Review by jtrindade — APPROVED** [review]
See the two comments.

File: README.md (hunk 2 above)
Comment (grimord) [reply README.md:54]: Added an explicit listing of available ports as suggested. They were only listed in the "Customizing" section which was, in retrospect, a mistake.

File: README.md (hunk 3 above)
Comment (grimord) [reply README.md]: Ended up rewritting this entire sentence as we don't actually **build** any images besides the Integration JAR, so my previous wording was unintentionally misleading.

**Review by jtrindade — APPROVED** [review]

**Pull request merged**